cognite/client/cognite_client.py

```diff
@@ -183,11 +189,40 @@ def delete(self, url: str, params: Dict[str, Any] = None, headers: Dict[str, Any
 
     def _client_factory(self, client):
         return client(
+            request_session=self._requests_session,
             project=self._project,
             base_url=self._base_url,
-            num_of_retries=self._num_of_retries,
             num_of_workers=self._num_of_workers,
             cookies=self._cookies,
             headers=self._headers,
             timeout=self._timeout,
         )
+
+    def _requests_retry_session(self):
+        session = Session()
+        retry = Retry(
+            total=self._num_of_retries,
+            read=self._num_of_retries,
+            connect=self._num_of_retries,
+            backoff_factor=1,
```

Comment:
```suggestion
            backoff_factor=0.5,
```
If I understand the documentation correctly - backoff_factor=1 means it will wait 0.0s, 2.0s, 4.0s, ...
Maybe it was intended to be 0.5 so that one gets 0.0s, 1.0s, 2.0s ...?